src/streaming/controllers/StreamController.js

```diff
@@ -144,28 +142,32 @@ function StreamController() {
         eventBus.on(Events.MANIFEST_UPDATED, onManifestUpdated, this);
         eventBus.on(Events.STREAM_BUFFERING_COMPLETED, onStreamBufferingCompleted, this);
         eventBus.on(MediaPlayerEvents.METRIC_ADDED, onMetricAdded, this);
-        if (skipDataLoss === true) {
-            eventBus.on(Events.WALLCLOCK_TIME_UPDATED, onWallclockTimeUpdated, this);
-        }
-
     }
 
 
     function onWallclockTimeUpdated(e) {
         wallclockTicked++;
-        if (wallclockTicked >= STALL_THRESHOLD_TO_CHECK_DATALOSS) {
+        if (wallclockTicked >= STALL_THRESHOLD_TO_CHECK_GAPS) {
             wallclockTicked = 0;
             let currentTime = playbackController.getTime();
             if (lastPlaybackTime === currentTime) {
-                skipWhenDataLoss(currentTime, e.timeToEnd);
+                skipGaps(currentTime, e.timeToEnd);
             } else {
                 lastPlaybackTime = currentTime;
             }
         }
 
     }
 
-    function skipWhenDataLoss(time, timeToStreamEnd) {
+    function setSkipGaps(value) {
+        if (value) {
+            eventBus.on(Events.WALLCLOCK_TIME_UPDATED, onWallclockTimeUpdated, this);
+        } else {
+            eventBus.off(Events.WALLCLOCK_TIME_UPDATED, onWallclockTimeUpdated, this);
```

Comment:
I think we should reset wallclockTicked here, right?

Reply:
We should also keep setting listener in initialize and remove it in reset. Otherwise skipGaps will stop being applied as soon as StreamController is restarted.